hw/ip_templates/rv_core_ibex/doc/boot-rom-patching.md

```diff
@@ -0,0 +1,476 @@
+
+This proposal introduces a second OpenTitan boot ROM partition with efuse/OTP based patching mechanism that enables each SoC to add such small customizations and provide critical survivability hooks while also removing the dependency on the integrated flash component.
+Integrated flash technology may not be available for SoCs built using advanced process nodes.
+
+Two models for OpenTitan based ROM patch deployment are envisioned:
+
+- **Factory patching** - here the ROM patches are programmed into the eFuses in the factory.
+  This may be to accommodate post-silicon bug fixes without needing a new step of the silicon.
+  Performed through the secure manufacturing flow.
+- **In-field patching** - The capability may be used to recover parts in the field but that requires an ability to burn eFuses in the field.
+
+Note that patch structure for both deployment models is the same.
+An infield patch can be viewed as a new revision of the OTP patch.
+The patch in eFuses will be signed using an appropriate PK signing algorithm
+
+### Programming method:
+
+- If prior to the PROD lifecycle stage: Using the manufacturing efuse programming interface.
+- In PROD lifecycle stage: Secure, verified patch update programming managed by secure application running on OpenTitan Ibex core (e.g., update patch revision in factory after provisioning, backup in-field patching mechanism).
+
+The factory provisioning model is expected to be the mainstream deployment model that covers bugs found throughout the product development lifecycle until product deployment phase.
+The in-field patching capability is primarily a backup mechanism used only in very rare cases to patch a critical ROM bug escape in the field, especially some hard to catch security or functional bugs.
+**It is advised to avoid reliance on the in-field ROM patch as much as possible.**
+
+This document also contains a detailed specification for the OpenTitan ROM patching architecture, the programming interfaces, and example sequences for patching ROM firmware.
+
+## ROM Boot & Patching Building Blocks
+
+![Integrated OpenTitan and Boot Building Blocks](building-blocks.svg)
+
+Following changes are envisioned to allow OpenTitan based boot customizations for an SoC to configure the access path to the external flash:
+
+### OpenTitan base ROM (first ROM partition)
+
+- Base ROM design is common across all integrated OpenTitan instances.
+
+- Supports security hardened functions such as signature verification, verification & loading of second ROM partition patch, other hardened crypto functions required at boot etc.
+  Integrated OpenTitan reuses the majority of current base ROM implementation.
+
+- Base ROM is not patchable.
+  Guaranteed to be fully verified and to be free of bugs.
+  To achieve this goal, it is kept simple by design.
+
+- Following is a summary of a few additional changes required (detailed out in later sections)
```

Comment:
Additional to what? The text doesn't quite work here.

My guess is that this is supposed to be saying "Once the majority of the base ROM is complete, it must run the following tasks before jumping to the second ROM partition" ?

(Minor nit: Also, I think the text in parens could just be "detailed in ...")